src/index.ts

```diff
@@ -83,695 +115,648 @@ if (!Object.keys) {
-  const hasAlt = (alt: string): boolean => {
-    return !!alt;
-  };
-  const sortAssets = (assets: Assets): void => {
-    const defAscend = (a: Asset, b: Asset): number => {
-      return a.index - b.index;
-    };
-
-    assets.sort(defAscend);
-  };
-  const toKebabCase = (str: string): string => {
-    return str
-      .replace(/([A-Z])([A-Z])/g, `$1-$2`)
-      .replace(/([a-z])([A-Z])/g, `$1-$2`)
-      .replace(/[\s_]+/g, `-`)
-      .toLowerCase();
-  };
-  const trimDir = (dir: string): string => {
-    let trimmedDir: string = dir;
-
-    trimmedDir = trimmedDir.replace(/\/{2,}/g, `/`);
-    trimmedDir = trimmedDir.replace(/(^\/*|\/*$)/g, ``);
-
-    return trimmedDir;
-  };
-
-  const sortedContexts: TShirtSizes = sortContexts(Object.keys(img.contexts));
-
-  for (let i: number = 0, len: number = sortedContexts.length; i < len; i++) {
-    const assets: Assets = img.contexts[sortedContexts[i]]!.assets;
-
-    sortAssets(assets);
 
-    if (img.compress) {
-      for (let i: number = 0, len: number = assets.length; i < len; i++) {
-        const args: AssetArgs = assets[i].args;
-
-        compressArg(`context`, args);
-        compressArg(`def`, args);
-        compressArg(`ext`, args);
-        compressArg(`qual`, args);
-        compressArg(`size`, args);
-      }
+      file.write(`\n`);
     }
-  }
 
-  const docName: string = app.activeDocument.name.replace(/\.[a-z]{3,4}$/i, ``);
-  const docPath: Folder = app.activeDocument.path;
-  const file: File = File(`${docPath}/${docName}-assets/responsive-image.html`);
-
-  if (file.exists) {
-    file.remove();
+    file.writeln(`${tabs}  "`);
   }
 
-  file.encoding = 'utf-8';
-  file.open('w');
-
-  const tab: string = `  `;
-
-  if (hasMultipleContexts(sortedContexts)) {
-    file.writeln(`<picture>`);
-
-    for (let i: number = sortedContexts.length; i > 0; i--) {
-      const sortedContext: TShirtSize = sortedContexts[i - 1];
-      const assets: Assets = img.contexts[sortedContext]!.assets;
-
-      if (i === 1) {
-        file.writeln(`${tab}<img`);
+  file.writeln(`${tabs}/>`);
+};
 
-        if (hasAlt(img.alt)) {
-          file.writeln(`${tab}${tab}alt="${img.alt}"`);
+/**
+ * Generates a picture tag from `resImg`, then writes it to `file`.
+ *
+ * @since 1.0.0
+ *
+ * @param {Object} file The file to write the img tag to.
+ * @param {Object} resImg The object containing alt text, src directory path and
+ * context max-widths.
+ * @returns {void} This function doesn't have a return statement.
+ */
+const generatePictureTag = (file: any, resImg: ResponsiveImage) => {
+  file.writeln(`<picture>`);
+
+  const contexts: AssetContext[] = sortContexts(Object.keys(assets));
+  const maxWidths: AssetContext[] = sortContexts(Object.keys(resImg.maxWidths));
+
+  for (let i: number = contexts.length; i > 0; i--) {
+    const contextualAssets: Asset[] = sortAssets(assets[contexts[i - 1]]!);
+    const minWidth: number = resImg.maxWidths[maxWidths[i - 2]]! + 1;
+
+    if (i > 1) {
```

Comment:
whats special about 1? is it 's' size?

Reply:
The two separate scenarios in the README.md should help explain the different behaviours here. If there is only one context, then an `<img>` tag is generated with relevant `src` and `srcset` attrs. Otherwise, `if (i > 1)` then a `<picture>` tag is generated with relevant `<source>` tags, and `src`, `srcset` and `media` attrs.

Reply:
But isnt _i_ here just the index of the current asset context being considered in the loop? Wouldnt the current code just generate source tags for each one of the other contexts but the one at index 0?

also, it would be good to start the loop with length - 1, then you wouldnt need to juggle with the i - 1 for the current index